pkg/kvstore/etcd_test.go

```diff
@@ -14,13 +14,91 @@ import (
 	"testing"
 	"time"
 
+	"github.com/cilium/cilium/pkg/testutils"
 	"github.com/stretchr/testify/require"
 	etcdAPI "go.etcd.io/etcd/client/v3"
 	"k8s.io/apimachinery/pkg/util/rand"
-
-	"github.com/cilium/cilium/pkg/testutils"
 )
 
+// SetupDummyWithConfigOpts sets up the dummy kvstore for tests but also
+// configures the module with the provided opts. A lock mechanism it used to
+// prevent the creation of two clients at the same time, to avoid interferences
+// in case different tests are run in parallel. A cleanup function is
+// automatically registered to delete all keys and close the client when the
+// test terminates.
+func SetupDummyWithConfigOpts(tb testing.TB, dummyBackend string, opts map[string]string) {
+	module := ExportedGetBackend(dummyBackend)
+	if module == nil {
+		tb.Fatalf("Unknown dummy kvstore backend %s", dummyBackend)
+	}
+
+	// Ensure the module is valid
+	if module == nil {
+		tb.Fatalf("Unknown dummy kvstore backend %s", dummyBackend)
+	}
+
+	// Perform a type assertion to get the concrete type (*EtcdModule)
+	etcdMod, ok := module.(*EtcdModule)
+	if !ok {
+		tb.Fatalf("Invalid module type, expected *EtcdModule")
+	}
+
+	// Call setConfigDummy on the concrete type
+	etcdMod.ExportedSetConfigDummy()
+
+	if opts != nil {
+		err := etcdMod.ExportedSetConfig(opts)
+		if err != nil {
+			tb.Fatalf("Unable to set config options for kvstore backend module: %v", err)
+		}
+	}
```

Comment:
I'd suggest trying to rework this snippet using `kvstore.NewClient` (or `kvstore.Setup`), which is already exported, and should avoid the need for exporting a bunch of dedicated test function (which may be even deleted afterwards).